R/plot_model_profile_survival.R

```diff
@@ -32,89 +36,415 @@
 #' plot(m_prof, numerical_plot_type = "contours")
 #'
 #' plot(m_prof, variables = c("trt", "age"), facet_ncol = 1)
+#' 
+#' plot(m_prof, geom = "variable", variables = "karno", plot_type = "pdp+ice")
+#' 
+#' plot(m_prof, geom = "variable", times = c(1, 2.72), variables = "karno", plot_type = "pdp+ice")
+#' 
+#' plot(m_prof, geom = "variable", times = c(1, 2.72), variables = "trt", plot_type = "pdp+ice")
 #' }
 #'
 #' @export
 plot.model_profile_survival <- function(x,
                                         ...,
+                                        geom = "time",
                                         variables = NULL,
                                         variable_type = NULL,
                                         facet_ncol = NULL,
                                         numerical_plot_type = "lines",
-                                        title = "Partial dependence survival profile",
+                                        times = NULL,
+                                        marginalize_over_time = FALSE,
+                                        plot_type = NULL,
+                                        title = "default",
                                         subtitle = "default",
                                         colors = NULL,
                                         rug = "all",
-                                        rug_colors = c("#dd0000", "#222222"))
-{
+                                        rug_colors = c("#dd0000", "#222222")) {
 
+    if (!geom %in% c("time", "variable")) {
+        stop("`geom` must be one of 'time' or 'survival'.")
+    }
+
+    if (geom == "variable") {
+
+        pl <- plot2(
+            x = x,
+            variable = variables,
+            times = times,
+            marginalize_over_time = marginalize_over_time,
+            plot_type = plot_type,
+            ... = ...,
+            title = title,
+            subtitle = subtitle,
+            colors = colors
+        )
+        return(pl)
+    }
 
+    lapply(list(x, ...), function(x) {
+        if (!inherits(x, "model_profile_survival")) {
+            stop("All ... must be objects of class `model_profile_survival`.")
+        }
+    })
     explanations_list <- c(list(x), list(...))
+    
     num_models <- length(explanations_list)
+    if (title == "default") {
+        if (x$type == "partial") {
+            title <- "Partial dependence survival profiles"
+        }
+        if (x$type == "accumulated") {
+            title <- "Accumulated local effects survival profiles"
+        }
+    }
```

Comment:
It would be nice to have separate default headers for ICE plots and PDP+ICE plots, otherwise default header for ICE plots is "Partial dependence survival profiles"
```suggestion
    num_models <- length(explanations_list)
    if (title == "default") {
  if (x$type == "partial") {
    if (geom == "variable") {
      if (plot_type == "ice") {
        title <- "Individual conditional expectation survival profiles"
      } else if (plot_type == "pdp+ice") {
        title <- "Partial dependence with individual conditional expectation survival profiles"}
    }
    title <- "Partial dependence survival profiles"
  }
  if (x$type == "accumulated") {
    title <- "Accumulated local effects survival profiles"
  }
}
```

Reply:
fully agree, fixed in [660141b](https://github.com/ModelOriented/survex/pull/82/commits/660141b5657f65ec70578c5b7706add9a29b6cb9)